src/crimson/common/errorator.h

```diff
@@ -728,6 +730,28 @@ struct errorator {
     }
   };
 
+  // assert_all{ "TODO" };
+  class assert_all {
+    const char* const msg = nullptr;
+  public:
+    template <std::size_t N>
+    assert_all(const char (&msg)[N])
+      : msg(msg) {
+    }
+    assert_all() = default;
+
+    template <class ErrorT, EnableIf<ErrorT>...>
+    decltype(auto) operator()(ErrorT&&) {
```

Comment:
why `decltype(auto)` here? because `assert_all()` can be plugged in to places where a return value is expected ? would `void` be a better option?

Reply:
`void` would do, I think. `discard_all` exhibits the same problem. Fixing.